huaweicloud/services/cse/resource_huaweicloud_cse_microservice.go

```diff
@@ -0,0 +1,226 @@
+)
+
+func ResourceMicroservice() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMicroserviceCreate,
+		ReadContext:   resourceMicroserviceRead,
+		DeleteContext: resourceMicroserviceDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMicroserviceImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"connect_address": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z0-9]([\w-.]*[A-Za-z0-9])?$`),
+						"The name must start and end with a letter or a digit, and can only contain letters, digits, "+
+							"underscore (_), hyphens (-) and dots (.)."),
+					validation.StringLenBetween(1, 128),
+				),
+			},
+			"app_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"environment": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"development", "testing", "acceptance", "production",
+				}, false),
+			},
+			"level": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"FRONT", "MIDDLE", "BACK",
+				}, false),
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 256),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "UP",
+				ValidateFunc: validation.StringInSlice([]string{
+					"UP", "DOWN",
+				}, false),
+			},
+			"admin_user": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"admin_pass": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Sensitive:    true,
+				ForceNew:     true,
+				RequiredWith: []string{"admin_user"},
+			},
+		},
+	}
+}
+
+func buildMicroserviceCreateOpts(d *schema.ResourceData) services.CreateOpts {
+	env := d.Get("environment").(string)
+	return services.CreateOpts{
+		Services: services.Service{
+			Name:        d.Get("name").(string),
+			AppId:       d.Get("app_name").(string),
+			Environment: &env,
+			Version:     d.Get("version").(string),
+			Level:       d.Get("level").(string),
+			Description: d.Get("description").(string),
+			Status:      d.Get("status").(string),
+		},
+	}
+}
+
+func resourceMicroserviceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var token string
+	if user, ok := d.GetOk("admin_user"); ok {
+		var err error
+		client := NewCustomClient(d.Get("connect_address").(string), "v4")
```

Comment:
why the version is 'v4' , not get from `version`

Reply:
This function is to generate the customized client